atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/TimestampStateStore.java

```diff
@@ -55,7 +59,7 @@
  *    caller.
  * 2. The entries in the living versions multimap may not be independent (as a single version may correspond to many
  *    timestamps), but the update concurrency should be handled by the data structure.
- * 3. Calls to {@link #getEarliestLiveSequence()} are synchronised on the livingVersions map, and thus are blocking;
+ * 3. Calls to {@link #getEarliestLiveSequence()} uses a sorted snapshot of livingVersions map;
```

Comment:
```suggestion
 * 3. Calls to {@link #getEarliestLiveSequence()} uses a snapshot of livingVersions map;
```

nit: I think we ended up with a ConcurrentHashMap, so the snapshot isn't sorted (but we can find its min through iteration)

Reply:
good catch, updated